src/StimulusBundle/tests/Dto/StimulusActionAttributeTest.php

```diff
@@ -0,0 +1,118 @@
+    }
+
+    /**
+     * @return iterable<array{
+     *     controllerName: string,
+     *     actionName: string,
+     *     eventName: ?string,
+     *     expectedAction: string,
+     * }>
+     */
+    public static function provideAddActionData(): iterable
+    {
+        // basic datasets
+        yield 'foo#bar' => [
+            'controllerName' => 'foo',
+            'actionName' => 'bar',
+            'eventName' => null,
+            'expectedAction' => 'foo#bar',
+        ];
+        yield 'baz->foo#bar' => [
+            'controllerName' => 'foo',
+            'actionName' => 'bar',
+            'eventName' => 'baz',
+            'expectedAction' => 'baz->foo#bar',
+        ];
+
+        // datasets from https://github.com/hotwired/stimulus
+        yield 'keydown.esc@document->a#log' => [
+            'controllerName' => 'a',
+            'actionName' => 'log',
+            'eventName' => 'keydown.esc@document',
+            'expectedAction' => 'keydown.esc&#x40;document->a#log',
+        ];
+        yield 'keydown.enter->a#log' => [
+            'controllerName' => 'a',
+            'actionName' => 'log',
+            'eventName' => 'keydown.enter',
+            'expectedAction' => 'keydown.enter->a#log',
+        ];
+        yield 'keydown.shift+a->a#log' => [
+            'controllerName' => 'a',
+            'actionName' => 'log',
+            'eventName' => 'keydown.shift+a',
+            'expectedAction' => 'keydown.shift&#x2B;a->a#log',
+        ];
+        yield 'keydown@window->c#log' => [
+            'controllerName' => 'c',
+            'actionName' => 'log',
+            'eventName' => 'keydown@window',
+            'expectedAction' => 'keydown&#x40;window->c#log',
+        ];
+        yield 'click->c#log:once' => [
+            'controllerName' => 'c',
+            'actionName' => 'log:once',
+            'eventName' => 'click',
+            'expectedAction' => 'click->c#log&#x3A;once',
+        ];
+
+        // extended datasets
+        yield 'vue:mount->foo#bar:passive' => [
+            'controllerName' => 'foo',
+            'actionName' => 'bar:passive',
+            'eventName' => 'vue:mount',
+            'expectedAction' => 'vue&#x3A;mount->foo#bar&#x3A;passive',
+        ];
+        yield 'foo--controller-1:baz->bar--controller-2#log' => [
+            'controllerName' => '@bar/controller_2',
+            'actionName' => 'log',
+            'eventName' => '@foo/controller_1:baz',
+            'expectedAction' => 'foo--controller-1&#x3A;baz->bar--controller-2#log',
+        ];
+        yield 'foo--controller-1:baz@document->bar--controller-2#log:capture' => [
+            'controllerName' => '@bar/controller_2',
+            'actionName' => 'log:capture',
+            'eventName' => '@foo/controller_1:baz@document',
+            'expectedAction' => 'foo--controller-1&#x3A;baz&#x40;document->bar--controller-2#log&#x3A;capture',
+        ];
+
+        // TODO: Add more datasets.
```

Comment:
It would be great if someone could suggest datasets that break the current implementation.

Reply:
Do you think there is many of them ? 😅

Reply:
Apparently this is enough 😁